addons/yatl/event/event.gd

```diff
@@ -43,3 +44,190 @@ const StreamOfflineEvent: Resource = preload("./types/events.gd").StreamOfflineE
+	if !__connection:
+		return
+
+	__time_since_last_message += delta
+
+
+# Pubic methods
+
+func connect_event(
+	_event: String,
+	_target: Object,
+	_method: String,
+	_binds: Array = []
+) -> int:
+	# TODO: Check if event is a valid event
+	# return 1
+
+	if !__connection && !yield(__establish_connection(__EVENT_WSS_URL), "completed"):
+		return 2 # Some kind of error
+
+	if __connections.has(_event):
+		__connections[_event].append([
+			funcref(_target, _method),
+			_binds,
+		])
+
+		return OK
+
+	# TODO: Convert this to use API method
+	var response = __pal.request(
+		'http://127.0.0.1:25708/helix/eventsub/subscriptions',
+		{
+			"client-id": __client_id,
+			"authorization": "Bearer %s" % __access_token,
+			"content-type": "application/json",
+		},
+		false,
+		HTTPClient.METHOD_POST,
+		to_json({
+			"type": _event,
+			"version": "1",
+			"condition": {
+				"broadcaster_user_id": __broardcaster_user_id,
+			},
+			"transport": {
+				"method": "websocket",
+				"id": __websocket_id,
+			},
+		})
+	)
+
+	response = yield(response, "completed")
+
+	if response.error:
+		return 3
+
+	if response.response_code != 200:
+		print(response.response_code)
+		print(response.body)
+		return 4
+
+	__connections[_event] = []
+	__connections[_event].append([
+		funcref(_target, _method),
+		_binds,
+	])
+
+	return OK
+
+
+# Private methods
+
+func __establish_connection(url: String) -> bool:
+	__connection = yield(__pal.establish_connection(url), "completed")
+
+	# TODO: Determine if reconnect should be tried
+	if !__connection.connected:
+		return false
+
+	# Wait for welcome message
+	var data_string: String = yield(__connection, "data_received")
+	var data: Dictionary = parse_json(data_string)
+
+	var type: String = data["metadata"]["message_type"]
+
+	if type != "websocket_welcome":
+		__connection.disconnect_socket()
+		__connection = null
+
+		return false
+
+	var websocket_data: Dictionary = data["payload"]["websocket"]
+
+	__websocket_id = websocket_data["id"]
+	__websocket_frequency_min = float(websocket_data["minimum_message_frequency_seconds"])
+
+	# Connect signal for future data
+	__connection.connect("data_received", self, "__data_received")
+
+	return true
+
+
+func __data_received(data_string: String) -> void:
+	var data: Dictionary =  parse_json(data_string)
+
+	# TODO: Replay detection / prevention, LRU queue
+
+	var type: String = data["metadata"]["message_type"]
+
+	match type:
+		"notification":
+			__handle_notification(data["payload"])
+
+	__time_since_last_message = 0.0
+
+
+func __handle_notification(payload: Dictionary) -> void:
+	var subscription_type: String = payload["subscription"]["type"]
+
+	var raw_event_data: Dictionary = payload["event"]
+
+	if __connections.has(subscription_type):
+		var name: String = "%sEvent" % subscription_type \
+			.replace(".", " ") \
+			.capitalize() \
+			.replace(" ", "") \
+			.replace("ChannelChannel", "Channel")
+		var event_resource: Resource = get(name)
```

Comment:
This is a Node, so you are shadowing its own `name` prop.

Reply:
Good catch. This will likely be replaced with a method so we can do an early check in the connect if we support the event.